### PR TITLE
Add new "golangci-lint" target to Makefile

### PR DESCRIPTION
**Pull request by thom-at-redhat**
# Description
In preparation for Go 1.18, we are adding a new "golangci-lint" target to Makefile allows running golangci-lint
Also minor reformatting in Makefile

Fixes # (issue)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Review by mgold1234 — COMMENTED** [review]
its look good. 
wander if there is a possibility to add somewhere a test that checks those changes

**Comment by thom-at-redhat** [comment]
> its look good. wander if there is a possibility to add somewhere a test that checks those changes
It's out of scope for this PR but I can see the following:

- syntax checking using something like checkmake
- functional testing to ensure all reasonable make targets complete successfully

File: Makefile [hunk 1]
```diff
@@ -95,6 +105,7 @@ help:
 	@echo "generate-docs             Creates OpenAPI specification for the project"
 	@echo "help                      Show this message"
 	@echo "lint                      Runs 'golint' on the project"
+	@echo "newlint					 Runs 'golangci-lint' on the project"
```
Comment (AaronH88) [inline Makefile:108]: lint and new lint are ambiguous names, I suggest change them to be more explicit, e.g. `Make golint` and `Make golangci-lint`

Comment (thom-at-redhat) [reply Makefile]: done

**Review by AaronH88 — APPROVED** [review]
/lgtm

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**